posthog/api/hog_function.py

```diff
@@ -0,0 +1,180 @@
+import structlog
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import serializers, viewsets
+from rest_framework.serializers import BaseSerializer
+
+from posthog.api.forbid_destroy_model import ForbidDestroyModel
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.api.shared import UserBasicSerializer
+from posthog.hogql.bytecode import create_bytecode
+from posthog.hogql.parser import parse_program
+from posthog.models.hog_functions.hog_function import HogFunction
+from posthog.models.hog_functions.utils import generate_template_bytecode
+from posthog.permissions import PostHogFeatureFlagPermission
+
+
+logger = structlog.get_logger(__name__)
+
+
+class InputsSchemaItemSerializer(serializers.Serializer):
+    type = serializers.ChoiceField(choices=["string", "boolean", "dictionary", "choice", "json"])
+    key = serializers.CharField()
+    label = serializers.CharField(required=False)
+    choices = serializers.ListField(child=serializers.DictField(), required=False)
+    required = serializers.BooleanField(default=False)
+    default = serializers.JSONField(required=False)
+    secret = serializers.BooleanField(default=False)
+    description = serializers.CharField(required=False)
+
+    # TODO Validate choices if type=choice
```

Comment:
Not sure if worth it today (this seems to be replicating the existing plugin logic somewhat), but I think a better approach than `type = ['string', 'bool', 'choice']` is `type = ['string', 'bool'], display = [none, 'choice']`, as we will want a choice between JSON nodes or integers one day.